src/common/components/select-repository-list/index.js

```diff
@@ -202,7 +190,16 @@ function mapStateToProps(state) {
     snaps,
     repositories,
     repositoriesStatus,
-    selectRepositoriesForm
+    selectRepositoriesForm,
+    isAddingSnaps: isAddingSnaps(state),
+    isRepoSelected: (repo) => hasRepository(selectRepositoriesForm.selectedRepos, repo),
```

Comment:
reselect memoized selectors don't support passing any parameters to them (they say all data needed for selector should be in the store), so to check particular repo we make some fake 'selectors' here.